services-directions-models/src/main/java/com/mapbox/api/directions/v5/utils/FormatUtils.java

```diff
@@ -0,0 +1,218 @@
+package com.mapbox.api.directions.v5.utils;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.mapbox.geojson.Point;
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Methods to convert models to Strings.
+ */
+public class FormatUtils {
+  /**
+   * Returns a string containing the tokens joined by delimiters.
+   *
+   * @param delimiter the delimeter on which to split.
+   * @param tokens A list of objects to be joined.. Strings will be formed from the objects by
+   *               calling object.toString().
+   * @return {@link String}
+   */
+  public static @Nullable String join(@NonNull CharSequence delimiter, @Nullable List<?> tokens) {
+    if (tokens == null || tokens.size() < 1) {
+      return null;
+    }
+
+    // trailing nulls should be dropped
```

Comment:
Are we sure that this is correct @korshaknn @Guardiola31337 @danpaz?

If I understand correctly, the nulls/empty strings should only be avoided for waypoints. For fields like `bearings` or `radiuses` we still want to have all of the `nulls`/empty strings be represented, even if they are trailing because the number of arguments has to much the number of coordinates.

Reply:
current code works with Strings and `split` is used o parse input string. 
`split` method removes trailing empty strings, so I just implemented the same logic, but with `Lists`
please check `MapboxDirectionsUtils`, to parse Strings we use Kotlin
`someString.split(SEMICOLON.toRegex()).dropLastWhile { it.isEmpty() }`
so, we drop empty strings. (it works exactly as java `split` does).

Reply:
Is that what Directions API expects though? The number of input values is going to differ in comparison to the size of the coordinate, isn't it?

Reply:
Yeah @LukasPaczos is right here 👇 

> If I understand correctly, the nulls/empty strings should only be avoided for waypoints. For fields like `bearings` or `radiuses` we still want to have all of the `nulls`/empty strings be represented, even if they are trailing because the number of arguments has to much the number of coordinates.

We need to make sure the number of coordinates match with the number of coordinates (except for `waypointIndices` as discussed above).

cc @danpaz